### DISCUSSION TITLE
Available tabs events?

### DISCUSSION THREAD

**Discussion by gouku** [facebook/docusaurus#5548] [Q&A] [answered]
Does [Tabs](https://docusaurus.io/docs/markdown-features/tabs) has any event available? So I can trigger other functions when specific tab shown or hidden.

Like how I did with [Bootstrap](https://getbootstrap.com/docs/5.1/components/navs-tabs/#events):

```js
tabEl.addEventListener('shown.bs.tab', function (event) {
  event.target // newly activated tab
  event.relatedTarget // previous active tab
})
```

**Comment by lex111** [comment]
No, tabs don't emit custom events, so if you want to change its behavior, you need to swizzle `Tabs` component .

**Reply by gouku** [reply]
Thanks for replying. Just learned how to swizzle and customize components. I know some JavaScript but pretty new to React. Any guide how to emit custom events for React components? Thanks!

**Comment by Josh-Cena** [comment, marked as answer]
The React way is actually to pass the behavior into `Tabs` as a callback:

```jsx
// Tabs.js
function Tabs(props) {
  const handleTabChange = (event) => {
    // ...
    props.onTabChange();
  }
  // ...
}

// your-page.{md,js}
<Tabs onTabChange={() => console.log("Tabs changed")} />
// ...
```

If you absolutely need to emit events, do it in the old way with DOM:

```jsx
function Tabs(props) {
  const ref = React.createRef();
  const handleTabChange = (event) => {
    // ...
    ref.current?.dispatchEvent(new Event('tabChange'));
  }
  return (
    <div ref={ref}>
       {/* the component */}
    </div>
  )
}
```

(Hope the latter works, correct me if it doesn't)

**Reply by gouku** [reply]
Thanks! The first approach just works for me!
